### DISCUSSION TITLE
Rancher Desktop Design - Images

### DISCUSSION THREAD

**Discussion by rak-phillip** [rancher-sandbox/rancher-desktop#695] [Archive]
## Images

Images is where a bulk of the proposed changes reside. The goal is to make all the different actions we can take while working with images a little more focused

### List Images

List all of the images, similar to how we do today with a few changes:

* Perform bulk actions, like scan, on multiple images
* Extract Image Acquisition as a child route for Images
* Scan results are viewable in as a child route

![2-images](https://user-images.githubusercontent.com/835961/134973939-ebd5a56a-776f-4b75-a44f-a4dd32d2b3fb.png)

![Image scan results   progress](https://user-images.githubusercontent.com/835961/134973989-d583fcd1-9ff1-4467-8b56-1b61c3614456.png)

### Image Acquisition

* Pull and Build actions are broken out into two separate tabs
* Output is rendered similar to how it is today

![3-images – add](https://user-images.githubusercontent.com/835961/134974844-d8a0302b-0427-4b15-8166-1623ebdb76d8.png)

![8-images – add – 1](https://user-images.githubusercontent.com/835961/134974868-e47a87cd-456a-4616-85ae-bf16bf3b985c.png)

### Image Scan Detail

* Output is parsed as table
* Ability to export results, if desired
* Link to CVE can be opened in browser window
* Any item can be expanded to view a description and list of references if they exist

![6-after-full-detail](https://user-images.githubusercontent.com/835961/134975248-0cc05390-4366-4748-a096-639bdf5327fd.png)

![6-after-full-detail – 1](https://user-images.githubusercontent.com/835961/134975275-e0ffef1f-cb7c-4614-a342-e0f8dacdaee5.png)

> Using the space to share this initial design proposal from the UI group. I've add some quick notes for each section of Rancher Desktop that might change. If you have any questions about anything you see below, please ask and I'll provide some more detail. 
>
> Original Discussion #683
>
> Feedback is highly valued and welcome.

**Comment by jandubois** [comment]
> ## Images
> Images is where a bulk of the proposed changes reside. The goal is to make all the different actions we can take while working with images a little more focused

I haven't thought through the "Images" functionality yet. But it is getting more complicated:

If we allow multiple runtimes concurrently (and I think this is what @mattfarina wants to disallow, but I disagree, and want to discuss this further), then we can end up with multiple distinct image repositories.

E.g. we might run Kubernetes with `containerd` but also run `moby` for docker-cli access. So we have some images managed by 'containerd` and some by `moby`. Or even we only use `containerd`, we may have Kubernetes images in the `k8s.io` namespace, and onces created by `nerdctl` in `default`, and the user may want to treat them separately.

So I don't know how this can be made natural in the UI. Potentially we need an image tab for each of the runtimes, even if they share much of the implementation.

I understand that disallowing multiple runtimes "solves" this problem, but I think there is value in being able to enable e.g. Podman to explore something new while keeping moby/docker working for your current projects. Let's discuss this maybe in person?

Comment copied from https://github.com/rancher-sandbox/rancher-desktop/discussions/683#discussioncomment-1393162

**Reply by rak-phillip** [reply]
What I'm hearing is that we need to get ahead of the current state. What we have right now are designs that will hopefully set a foundation for some of our more basic interactions in Rancher Desktop. I agree, let's discuss this in person.
